### PR TITLE
bartender: add `passthru.updateScript`, format with `nixfmt-rfc-style`, use `lib.platforms.darwin`, 5.2.3 -> 5.2.7

### PR DESCRIPTION
**Pull request by 4evy**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ba/bartender/package.nix [hunk 1]
```diff
@@ -32,17 +39,37 @@ stdenvNoCC.mkDerivation (finalAttrs: {
     runHook postInstall
   '';
 
+  passthru.updateScript = lib.getExe (writeShellApplication {
+    name = "bartender-update-script";
+    runtimeInputs = [
+      curl
```
Comment (getchoo) [inline pkgs/by-name/ba/bartender/package.nix:45]:
```suggestion
      cacert
      curl
```
*Pretty sure* we also need this if running the script purely?

File: pkgs/by-name/ba/bartender/package.nix [hunk 2]
```diff
@@ -32,17 +39,37 @@ stdenvNoCC.mkDerivation (finalAttrs: {
...
+      xmlstarlet
+      common-updater-scripts
+    ];
+    text = ''
+      version_major="${lib.versions.major finalAttrs.version}"
+      url="https://www.macbartender.com/B2/updates/AppcastB$version_major.xml"
+      version=$(curl -s "$url" | xmlstarlet sel -t -v '(//item)[last()]/sparkle:shortVersionString' -n)
+      update-source-version bartender "$version" --file=./pkgs/by-name/ba/bartender/package.nix
```
Comment (getchoo) [inline pkgs/by-name/ba/bartender/package.nix:53]:
```suggestion
      update-source-version bartender "$version"
```
`update-source-version` *should* also be able to find this on it's own. Correct me if I'm wrong though :p

**Review by getchoo — COMMENTED** [review]
Otherwise LGTM

File: pkgs/by-name/ba/bartender/package.nix (hunk 1 above)
Comment (DontEatOreo) [reply pkgs/by-name/ba/bartender/package.nix:45]: From my understanding update scripts are suppose to be ran with https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/update.nix *(e.g `APP=bartender; nix-shell --pure ./maintainers/scripts/update.nix --argstr package "$APP" --argstr commit true --argstr skip-prompt true`)*

And in `update.nix` we have this line: https://github.com/NixOS/nixpkgs/blob/0311f6c40e0a7ea6bb1e967b5a607871c654d5f6/maintainers/scripts/update.nix#L234

Which gives us the cacert already

File: pkgs/by-name/ba/bartender/package.nix (hunk 2 above)
Comment (DontEatOreo) [reply pkgs/by-name/ba/bartender/package.nix:53]: Ahh, good catch! I remember at one point it wasn't possible; it seems like it's not the case anymore

File: pkgs/by-name/ba/bartender/package.nix [hunk 3]
```diff
@@ -32,17 +39,37 @@ stdenvNoCC.mkDerivation (finalAttrs: {
...
+    }/rnotes.html";
     license = [ lib.licenses.unfree ];
     sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];
-    maintainers = with lib.maintainers; [ stepbrobd DimitarNestorov ];
-    platforms = [ "aarch64-darwin" "x86_64-darwin" ];
+    maintainers = with lib.maintainers; [
+      stepbrobd
+      DimitarNestorov
+    ];
+    platforms = lib.platforms.darwin;
```
Comment (stepbrobd) [inline pkgs/by-name/ba/bartender/package.nix:73]:
```nix
nix-repl> lib.platforms.darwin
[
  "x86_64-darwin"
  "i686-darwin"
  "aarch64-darwin"
  "armv7a-darwin"
]
```

I don't think bartender supports i686-darwin and armv7a-darwin

Comment (DontEatOreo) [reply pkgs/by-name/ba/bartender/package.nix:73]: Yea, but on nix search page it just shows x86 and arch64 only, plus there's no Apple PC devices that have these architectures...

This is more for syntax sugar than anything else

Comment (stepbrobd) [reply pkgs/by-name/ba/bartender/package.nix:73]: idk what the current convention is, if `lib.platforms.darwin` is preferred over explicit listings in other darwin packages then feel free to ignore this

Comment (DontEatOreo) [reply pkgs/by-name/ba/bartender/package.nix:73]: Idk if there's any convention, but from a POV, it's "cleaner" to have `lib.platforms.darwin`. As well, it's impossible to run Nix on `armv7a-darwin` (32-bit iPhones?) or `i686-darwin` (OS X 10.3?), so it's impossible to have a situation where someone will be trying to build Bartender on those platforms.

Comment (getchoo) [reply pkgs/by-name/ba/bartender/package.nix:73]: I'd say this is fine, especially as I don't think anything *technically* stops it? I've also only rarely seen an explicit listing of x86_64/aarch64-darwin

Maybe use [meta.badPlatforms](https://nixos.org/manual/nixpkgs/stable/#var-meta-badPlatforms)?

File: pkgs/by-name/ba/bartender/package.nix (hunk 1 above)
Comment (getchoo) [reply pkgs/by-name/ba/bartender/package.nix:45]: > Correct me if I'm wrong, but from my understanding, update scripts are supposed to be ran with https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/update.nix

They can be, but they can (or well, should) also be run on their own. Sorta a nitpick, but I think it's a nice thing to account for

**Review by getchoo — APPROVED** [review]
Feel free to ping in a couple days if there are no other reviews

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 363633`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bartender</li>
  </ul>
</details>

**Comment by DimitarNestorov** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 363633`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>bartender</li>
  </ul>
</details>

@NixOS/nixpkgs-merge-bot merge

**Review by DimitarNestorov — APPROVED** [review]

**Pull request merged**